### PR TITLE
Add logging

### PR DESCRIPTION
**Pull request by enyst**
This is WIP rebased on top of the switch to poetry branch. Sorry it's harder to see diff this way; that will be fixed.

@dorbanianas introduced logging on the testing component, this PR brings it in the opendevin module.

Introduce logging, and start replacing print() with logs.
- general logger ./logs/opendevin.log
- also log in console
- special logger for LLM: keeps separate LLM debug logs like we currently have, one file per prompt and one per response, in ./logs/llm
- (Note to self) log json_repair when it did something.

### CODE REVIEW

File: opendevin/_logging.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import logging, os, sys
```
Comment (rbren) [inline opendevin/_logging.py:1]: Is the underscore prefix for the filename typical of python? I haven't seen that before

**Review by rbren — APPROVED** [review]
This is rad. Thanks for doing this!

File: opendevin/_logging.py (hunk 1 above)
Comment (enyst) [reply opendevin/_logging.py:1]: LiteLLM is doing it, among our stuff, and I stole it. But it's very uncommon. PEP 8 doesn't mention it directly, and at most suggests that it's native code (c/c++). I'll change it.

File: opendevin/_logging.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    console_handler.setFormatter(console_formatter)
+    return console_handler
+
+def get_file_handler():
+    """
+    Returns a file handler for logging.
+    """
+    log_dir = os.path.join(os.getcwd(), "logs")
+    os.makedirs(log_dir, exist_ok=True)
+    file_handler = logging.FileHandler(os.path.join(log_dir, "opendevin.log"))
```
Comment (dorbanianas) [inline opendevin/_logging.py:32]:
```suggestion
    timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
    file_name = f"opendevin_{timestamp}.log"
    file_handler = logging.FileHandler(os.path.join(log_dir, file_name ))
```

Adding the timestamp would be better just for visibility and following the changes over each execution, what do you think?

Comment (enyst) [reply opendevin/_logging.py]: Absolutely, thank you!

Comment (dorbanianas) [reply opendevin/_logging.py]: I guess you should `import datetime`

**Comment by neubig** [comment]
This has been approved, so let's resolve the conflicts and get it merged in!

**Review by dorbanianas — APPROVED** [review]

**Pull request merged**